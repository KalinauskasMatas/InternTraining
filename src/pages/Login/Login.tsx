import React from "react";

import Header from "../../components/Header/Header";
import AuthForm from "../../components/AuthForm/AuthForm";
import Footer from "../../components/Footer/Footer";

const Login = () => {
  return (
    <div className="body-wrapper">
      <Header title={"Movie rental"} />
      <AuthForm isRegister={false} />
      <AuthForm isRegister={true} />
      <Footer />
    </div>
  );
};

export default Login;
