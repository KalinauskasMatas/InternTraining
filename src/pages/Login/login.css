.login-form,
.register-form {
    background-color: var(--bright-background-color);
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    width: 250px;
    margin: auto;
    margin-bottom: 15px;
    border-radius: 30px;
}

.register-form {
    display: none;
}

.login-form > label,
.register-form > label {
    margin: 15px 30px 3px 30px;
    font-size: 10px;
}

a {
    margin: auto;
}

form > input,
form > input {
    margin: 0 30px 0 30px;
    border: none;
    background-color: var(--background-color);
    padding: 5px 10px;
    align-self: stretch;
}

.form-button {
    width: 125px;
    margin: 10px auto 12px auto;
    background-color: var(--dark-button-color);
    color: var(--bright-text-color);
    border: none;
    padding: 5px 35px;
    border-radius: 15px;
    cursor: pointer;
}