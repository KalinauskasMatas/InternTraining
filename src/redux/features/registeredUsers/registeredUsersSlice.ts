import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { UserState } from "../../../interfaces";
import { RootState } from "../../store";

const initialState: UserState[] =
  JSON.parse(localStorage.getItem("registeredUsers") as string) || [];

const registeredUsersSlice = createSlice({
  name: "registeredUsers",
  initialState,
  reducers: {
    registerUser: (state, action: PayloadAction<UserState>) => {
      state.push({
        fname: action.payload.fname,
        surname: action.payload.surname,
        email: action.payload.email,
        password: action.payload.password,
        rentMovies: action.payload.rentMovies,
      });

      localStorage.setItem("registeredUsers", JSON.stringify(state));
    },

    updateUser: (state, action: PayloadAction<UserState>) => {
      const userIndex = state.findIndex(
        (user) => user.email === action.payload.email
      );
      if (userIndex === -1) return;
      state[userIndex] = { ...action.payload };

      localStorage.setItem("registeredUsers", JSON.stringify(state));
    },

    usersSetEmail: (
      state,
      action: PayloadAction<{ user: UserState; newEmail: string }>
    ) => {
      const userIndex = state.findIndex(
        (user) => user.email === action.payload.user.email
      );

      if (userIndex === -1) return;
      state[userIndex] = {
        ...state[userIndex],
        email: action.payload.newEmail,
      };

      localStorage.setItem("registeredUsers", JSON.stringify(state));
    },
  },
});

export const { registerUser, updateUser, usersSetEmail } =
  registeredUsersSlice.actions;

export const selectRegisteredUsers = (state: RootState) =>
  state.registeredUsers;

export default registeredUsersSlice.reducer;
