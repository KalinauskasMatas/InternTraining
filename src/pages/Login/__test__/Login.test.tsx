import Login from "../Login";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";

import renderer from "react-test-renderer";

import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import registeredUsersSlice from "../../../redux/features/registeredUsers/registeredUsersSlice";

afterEach(cleanup);

const store = configureStore({
  reducer: {
    registeredUsers: registeredUsersSlice,
  },
});

it("renders without crashing", () => {
  render(
    <Provider store={store}>
      <Login />
    </Provider>
  );
});

it("matches with snapshot", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Login />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
