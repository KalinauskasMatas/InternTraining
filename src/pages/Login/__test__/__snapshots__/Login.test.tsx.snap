// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches with snapshot 1`] = `
<div
  className="body-wrapper"
>
  <header
    className="header-name"
    data-testid="header-title"
  >
    Movie rental
  </header>
  <article>
    <form
      className="auth-form"
      onSubmit={[Function]}
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        id="email"
        onChange={[Function]}
        placeholder="email"
        required={true}
        type="email"
      />
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        id="password"
        onChange={[Function]}
        placeholder="password"
        required={true}
        type="password"
      />
      <input
        className="form-button"
        type="submit"
        value="Sign In"
      />
      <p
        className="error"
      />
    </form>
  </article>
  <article>
    <button
      className="form-button"
      onClick={[Function]}
    >
      Register
    </button>
  </article>
  <footer>
    <span>
      Made by: Matas Kalinauskas
    </span>
    <span>
      2023
    </span>
  </footer>
</div>
`;
